extensions/sql-database-projects/src/test/netCoreTool.test.ts

```diff
@@ -41,8 +44,33 @@ describe('NetCoreTool: Net core install popup tests', function (): void {
 			should(result).true('dotnet is either not present or in /usr/local/share by default');
 		}
 	});
-});
 
-export async function sleep(ms: number): Promise<{}> {
-	return new Promise(resolve => setTimeout(resolve, ms));
-}
+	it('Should run dotnet command', async function (): Promise<void> {
+		this.timeout(10000 * 10); // higher timeout for this test
+
+		const netcoreTool = new NetCoreTool();
+		const projfile = path.join(__dirname, 'baselines', 'simpleProject.sqlproj');
+		const dacpacfile = path.join(__dirname, 'baselines', 'bin', 'debug', 'simpleProject.dacpac');
+
+		const buildHelper = new BuildHelper();
+		await buildHelper.createBuildDirFolder();
+		const arg = buildHelper.constructBuildArguments(projfile, buildHelper.extensionBuildDirPath);
+
+		const options: DotNetCommandOptions = {
+			argument: arg
+		};
+
+		try {
+			await netcoreTool.runDotnetCommand(options);
+			should(fs.existsSync(dacpacfile)).equal(true); //if net core is present
```

Comment:
any reason in particular not to use async?  probably should generally be the default for forward compat's sake

Reply:
https://nodejs.org/api/fs.html#fs_fs_promises_api Use this instead

Reply:
not sure why that would help as to we do have to wait for test completion on this. I don't mind changing bt not sure why we must.

Reply:
Because if you do a sync call it blocks the thread and prevents the engine from executing anything else until the call is finished. Using promises means that the engine can execute other tasks (such as running other async tasks) while it's waiting for the call to finish. 

For a test yes it's not super important, but it's best just to get in the habit of doing this all the time and avoid using sync methods completely (especially if the file system/network is involved)